### PR TITLE
Added new operator to check IP within range

### PR DESCRIPTION
**Pull request by sanket-mundra**
## Description
Please include a summary of the change, motivation and context.

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

### Checklist:
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

File: hypertrace-graphql-labels-schema-api/src/main/java/org/hypertrace/graphql/label/schema/rule/StringCondition.java [hunk 1]
```diff
@@ -14,7 +14,8 @@ public interface StringCondition {
   @GraphQLName(Operator.TYPE_NAME)
   enum Operator {
     OPERATOR_EQUALS,
-    OPERATOR_MATCHES_REGEX;
+    OPERATOR_MATCHES_REGEX,
+    OPERATOR_CIDR;
```
[the remark this answers is not in the document]
Comment (sanket-mundra) [reply hypertrace-graphql-labels-schema-api/src/main/java/org/hypertrace/graphql/label/schema/rule/StringCondition.java:18]: Will we have a use case where if certain APIs don't fall under given CIDR, we will add another/counter label to them?
cc @skjindal93

**Review by aaron-steinfeld — APPROVED** [review]

**Comment by sanket-mundra** [comment]
Just realised I have to work on converters as well 😅

**Review by aaron-steinfeld — CHANGES_REQUESTED** [review]
removing review

File: hypertrace-graphql-labels-schema-api/src/main/java/org/hypertrace/graphql/label/schema/rule/StringCondition.java [hunk 2]
```diff
@@ -9,12 +9,14 @@ public interface StringCondition {
   String TYPE_NAME = "LabelApplicationStringCondition";
 
   String OPERATOR_KEY = "operator";
-  String VALUE_KEY = "value";
+  String STRING_CONDITION_VALUE_KEY = "stringConditionValue";
```
Comment (aaron-steinfeld) [inline hypertrace-graphql-labels-schema-api/src/main/java/org/hypertrace/graphql/label/schema/rule/StringCondition.java:12]: Please revert this, it's a breaking change. Instead, accept a list of values as a sibling to existing value and remove the non null on `value` (still technically breaking, but less critical as it'd only break on trying to read a new value with an old query).

You may also want to include a nullable enum that specifies which field is set - your `StringConditionValueType`. It should always be set on return, but on request you would need to default it to the singular form for backwards compat. Hope that makes sense

File: hypertrace-graphql-labels-schema-api/src/main/java/org/hypertrace/graphql/label/schema/rule/StringCondition.java [hunk 3]
```diff
@@ -3,28 +3,51 @@
...
+  }
+
   @GraphQLField
   @GraphQLNonNull
   @GraphQLName(OPERATOR_KEY)
   Operator operator();
 
   @GraphQLField
-  @GraphQLNonNull
+  @Nullable
```
Comment (aaron-steinfeld) [inline hypertrace-graphql-labels-schema-api/src/main/java/org/hypertrace/graphql/label/schema/rule/StringCondition.java:40]: Another alternative would be to have value be typed Object and accept either a string or list of strings in it. We're not consistent, we've taken both approaches in different places. Up to you

File: hypertrace-graphql-labels-schema-impl/src/main/java/org/hypertrace/graphql/label/deserialization/LabelApplicationRuleDeserializationConfig.java [hunk 4]
```diff
@@ -146,6 +146,12 @@ private static class StringConditionArgument implements StringCondition {
 
     @JsonProperty(VALUE_KEY)
     String value;
+
+    @JsonProperty(VALUES_KEY)
```
Comment (aaron-steinfeld) [inline hypertrace-graphql-labels-schema-impl/src/main/java/org/hypertrace/graphql/label/deserialization/LabelApplicationRuleDeserializationConfig.java:150]: I guess it's pre-existing, but why did this class get implemented twice? Seems like a mistake here.

File: hypertrace-graphql-labels-schema-api/src/main/java/org/hypertrace/graphql/label/schema/rule/StringCondition.java (hunk 3 above)
Comment (sanket-mundra) [reply hypertrace-graphql-labels-schema-api/src/main/java/org/hypertrace/graphql/label/schema/rule/StringCondition.java:40]: If no harm in this approach, will keep it this way.

File: hypertrace-graphql-labels-schema-impl/src/main/java/org/hypertrace/graphql/label/deserialization/LabelApplicationRuleDeserializationConfig.java (hunk 4 above)
Comment (sanket-mundra) [reply hypertrace-graphql-labels-schema-impl/src/main/java/org/hypertrace/graphql/label/deserialization/LabelApplicationRuleDeserializationConfig.java:150]: `private static class LabelApplicationRuleArgument` is missing in one of them. Should I remove one of them, the one without `LabelApplicationRuleArgument` class?

**Review by aaron-steinfeld — APPROVED** [review]

**Pull request merged**